src/components/state/loading.tsx

```diff
@@ -1,29 +1,19 @@
-import { Component, h, Prop } from '@stencil/core';
-import { loading } from '../../utils/icon';
+import { FunctionalComponent, h } from '@stencil/core';
```

Comment:
Another possible structure would be to move this file inside `/player` and name it `states.tsx.` From there we can export the `loading` state and add others in the future if needed.
It is how we do it for the `controls` component and all its functional components (`elements.tsx`).

Not a blocker, just an idea... also fine with your approach.

Reply:
I like that approach. We can take it even further: We also name it `element.tsx` and include video and controls content there. I would do the latter in a follow-up refactoring. This way, we can further reduce the size of the far to long `player.tsx` class.